libs/core/src/components/pds-avatar/pds-avatar.tsx

```diff
@@ -79,13 +82,13 @@ export class PdsAvatar {
       this.dropdown
         ?
         <button class="pds-avatar__button" type="button">
-          <div style={style}>
+          <div style={style} part="image">
             {this.renderIconOrImage()}
             {this.renderBadge()}
           </div>
```

Comment:
This could be extracted to a method, e.g `iconContent`. This would remove the duplicate markup used in lines 91-94, which would be replaced with the call to the method you created.